src/Templates/InterfacesTemplate.cshtml

```diff
@@ -0,0 +1,81 @@
+@using AutoRest.Core.Utilities
+@using AutoRest.Go
+@using AutoRest.Go.Model
+@using AutoRest.Go.Templates
+@using System;
+@using System.Collections.Generic;
+@using System.Linq;
+
+@inherits AutoRest.Core.Template<AutoRest.Go.Model.CodeModelGo>
+
+@{
+    var content = new Dictionary<string, IEnumerable<MethodGo>>();
+    var imports = new HashSet<string> { PrimaryTypeGo.GetImportLine(package: "context") };
+
+    if (Model.ClientMethods.Any())
+    {
+        content.Add(Model.BaseClient, Model.ClientMethods.OrderBy(m => m.Name.Value));
+    }
+
+    var methodGroups = Model.MethodGroups.Where(mg => !string.IsNullOrEmpty(mg.Name));
```

Comment:
not sure why is this necessary ? if it is can you add a comment or maybe a method in the Model itself ?

Reply:
So `Model.MethodGroups` will contain all method groups including the ones in `Model.ClientMethods` which are just unnamed method groups.